cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -431,7 +431,9 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
             if (!this.isValidTimeLimit(this.$('.field-time-limit input').val())) {
                 this.$('.field-time-limit input').val('00:30');
             }
-            if (showRulesField) {
+
+            let  showReviewBlock = this.model.get('show_review_rules');
+            if (showRulesField && showReviewBlock) {
```

Comment:
this name isn't terrifically helpful in understanding the semantics underlying this. How do we understand the difference between the "rules field" and the "review block" these two variables refer to? Are they the same thing? It seems like it, since they control the same things.

Reply:
Open to name suggestions

Reply:
I agree. I was confused by the difference between the rules field and the review block. Maybe `showReviewBlock` could be `showReviewRulesForProvider`?

I feel that introducing this logic here violates the expectations that one has about the purpose of the `showRulesField` parameter. If I pass `True` or `False`, I can no longer be certain whether the rules field will show up, because we have introduced this additional condition for whether or not to show it. What do you think? I think fixing that would require more work.

Reply:
I agree we shouldn't populate something called `showReviewBlock` until we have this piece of information, since it's part of determining whether we should show the review block. If we were in python I'd be happy to `&=` the condition we're checking here into the current value, but in JS that sort of mutation doesn't match the vibe; it's better to choose good names and keep variables immutable.

Reply:
But to be clear I absolutely think this is worth fixing and shouldn't be a substantial amount more work.